From e45d1843451f02b71c0d7cd8db0ef475201f5221 Mon Sep 17 00:00:00 2001
From: Ivan Evlogiev <ivanevlogiev@codeaurora.org>
Date: Wed, 18 Sep 2013 17:12:02 +0300
Subject: [PATCH 014/176] QCamera2: Fix thumbnail reprocess skipping for HDR

- Skip thumbnail stream reprocessing for HDR, if only HDR is enabled

Change-Id: I3c0315c5a7045cdbb7206bed31fddb1253d65d61
---
 QCamera2/HAL/QCameraChannel.cpp | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/QCamera2/HAL/QCameraChannel.cpp b/QCamera2/HAL/QCameraChannel.cpp
index 313ca6a..1332b2e 100644
--- a/QCamera2/HAL/QCameraChannel.cpp
+++ b/QCamera2/HAL/QCameraChannel.cpp
@@ -720,11 +720,17 @@ int32_t QCameraReprocessChannel::addReprocStreamsFromSource(QCameraAllocator& al
                pStream->isTypeOf(CAM_STREAM_TYPE_POSTVIEW) ||
                pStream->isOrignalTypeOf(CAM_STREAM_TYPE_PREVIEW) ||
                pStream->isOrignalTypeOf(CAM_STREAM_TYPE_POSTVIEW)) {
-                  if (param.isHDREnabled() && !param.isHDRThumbnailProcessNeeded()){
+                  uint32_t feature_mask = config.feature_mask;
+
+                  if ((feature_mask & ~CAM_QCOM_FEATURE_HDR) == 0
+                      && param.isHDREnabled()
+                      && !param.isHDRThumbnailProcessNeeded()) {
+
                       // Skip thumbnail stream reprocessing in HDR
+                      // if only hdr is enabled
                       continue;
                   }
-                  uint32_t feature_mask = config.feature_mask;
+
                   //Don't do WNR for thumbnail
                   feature_mask &= ~CAM_QCOM_FEATURE_DENOISE2D;
                   if(!feature_mask) {
@@ -764,7 +770,14 @@ int32_t QCameraReprocessChannel::addReprocStreamsFromSource(QCameraAllocator& al
                 streamInfo->reprocess_config.pp_feature_config.feature_mask &= ~CAM_QCOM_FEATURE_CAC;
                 //Don't do WNR for thumbnail
                 streamInfo->reprocess_config.pp_feature_config.feature_mask &= ~CAM_QCOM_FEATURE_DENOISE2D;
+
+                if (param.isHDREnabled()
+                  && !param.isHDRThumbnailProcessNeeded()){
+                    streamInfo->reprocess_config.pp_feature_config.feature_mask
+                      &= ~CAM_QCOM_FEATURE_HDR;
+                }
             }
+
             if (streamInfo->reprocess_config.pp_feature_config.feature_mask & CAM_QCOM_FEATURE_ROTATION) {
                 if (streamInfo->reprocess_config.pp_feature_config.rotation == ROTATE_90 ||
                     streamInfo->reprocess_config.pp_feature_config.rotation == ROTATE_270) {
-- 
1.8.3.1

